### PR TITLE
Optimize AnteHandle method to skip checks if disabledMsgs is empty

### PR DESCRIPTION
**Pull request by fx0x55**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by mmsqe — APPROVED** [review]

**Comment by mmsqe** [comment]
Hi @fx0x55, can you help sign the first commit and add a change log?

**Comment by fx0x55** [comment]
> Hi @fx0x55, can you help sign the first commit and add a change log?

OK, done

File: CHANGELOG.md [hunk 1]
```diff
@@ -81,6 +81,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * [#496](https://github.com/crypto-org-chain/cronos/pull/496) Set mempool MaxTx from config.
 * (ante) [#497](https://github.com/crypto-org-chain/ethermint/pull/497) Enforce positive value check in eth transaction.
 * (deps) [#505](https://github.com/crypto-org-chain/ethermint/pull/505) Update cometbft to v0.38.10.
+* (ante) [#504](https://github.com/crypto-org-chain/ethermint/pull/505) Optimize AnteHandle method to skip checks if disabledMsgs is empty
```
Comment (mmsqe) [inline CHANGELOG.md]:
```suggestion
* (ante) [#504](https://github.com/crypto-org-chain/ethermint/pull/504) Optimize AnteHandle method to skip checks if disabledMsgs is empty.
```

**Comment by mmsqe** [comment]
> > Hi @fx0x55, can you help sign the first commit and add a change log?
> 
> OK, done

Thanks, can you sign all the commits? 
Screenshot 2024-08-02 at 11 28 55 AM

**Comment by fx0x55** [comment]
> > > Hi @fx0x55, can you help sign the first commit and add a change log?
> > 
> > 
> > OK, done
> 
> Thanks, can you sign all the commits? Screenshot 2024-08-02 at 11 28 55 AM

@mmsqe Ok, done

**Pull request merged**